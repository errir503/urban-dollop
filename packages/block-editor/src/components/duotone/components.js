/**
 * WordPress dependencies
 */
import { SVG } from '@wordpress/components';

/**
 * Internal dependencies
 */
import { __unstableGetValuesFromColors as getValuesFromColors } from './index';

/**
 * SVG and stylesheet needed for rendering the duotone filter.
 *
 * @param {Object} props          Duotone props.
 * @param {string} props.selector Selector to apply the filter to.
 * @param {string} props.id       Unique id for this duotone filter.
 *
 * @return {WPElement} Duotone element.
 */
export function DuotoneStylesheet( { selector, id } ) {
	const css = `
${ selector } {
	filter: url( #${ id } );
}
`;
	return <style>{ css }</style>;
}

/**
 * Stylesheet for disabling a global styles duotone filter.
 *
 * @param {Object} props          Duotone props.
 * @param {string} props.selector Selector to disable the filter for.
 *
 * @return {WPElement} Filter none style element.
 */
export function DuotoneUnsetStylesheet( { selector } ) {
	const css = `
${ selector } {
	filter: none;
}
`;
	return <style>{ css }</style>;
}

/**
 * The SVG part of the duotone filter.
 *
 * @param {Object}   props        Duotone props.
 * @param {string}   props.id     Unique id for this duotone filter.
 * @param {string[]} props.colors Color strings from dark to light.
 *
 * @return {WPElement} Duotone SVG.
 */
export function DuotoneFilter( { id, colors } ) {
	const values = getValuesFromColors( colors );
	return (
		<SVG
			xmlnsXlink="http://www.w3.org/1999/xlink"
			viewBox="0 0 0 0"
			width="0"
			height="0"
			focusable="false"
			role="none"
			style={ {
				visibility: 'hidden',
				position: 'absolute',
				left: '-9999px',
				overflow: 'hidden',
			} }
		>
			<defs>
				<filter id={ id }>
					<feColorMatrix
						// Use sRGB instead of linearRGB so transparency looks correct.
						colorInterpolationFilters="sRGB"
						type="matrix"
						// Use perceptual brightness to convert to grayscale.
						values="
							.299 .587 .114 0 0
							.299 .587 .114 0 0
							.299 .587 .114 0 0
							.299 .587 .114 0 0
						"
					/>
					<feComponentTransfer
						// Use sRGB instead of linearRGB to be consistent with how CSS gradients work.
						colorInterpolationFilters="sRGB"
					>
						<feFuncR
							type="table"
							tableValues={ values.r.join( ' ' ) }
						/>
						<feFuncG
							type="table"
							tableValues={ values.g.join( ' ' ) }
						/>
						<feFuncB
							type="table"
							tableValues={ values.b.join( ' ' ) }
						/>
						<feFuncA
							type="table"
							tableValues={ values.a.join( ' ' ) }
						/>
					</feComponentTransfer>
					<feComposite
						// Re-mask the image with the original transparency since the feColorMatrix above loses that information.
						in2="SourceGraphic"
						operator="in"
					/>
				</filter>
			</defs>
		</SVG>
	);
}

/**
 * SVG from a duotone preset
 *
 * @param {Object} props        Duotone props.
 * @param {Object} props.preset Duotone preset settings.
 *
 * @return {WPElement} Duotone element.
 */
export function PresetDuotoneFilter( { preset } ) {
	return (
		<DuotoneFilter
			id={ `wp-duotone-${ preset.slug }` }
			colors={ preset.colors }
		/>
	);
}
