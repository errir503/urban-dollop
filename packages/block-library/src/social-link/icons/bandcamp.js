/**
 * WordPress dependencies
 */

import { Path, SVG } from '@wordpress/primitives';

export const BandcampIcon = () => (
	<SVG width="24" height="24" viewBox="0 0 24 24" version="1.1">
		<Path d="M15.27 17.289 3 17.289 8.73 6.711 21 6.711 15.27 17.289" />
	</SVG>
);
