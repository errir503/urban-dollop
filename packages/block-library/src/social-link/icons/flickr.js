/**
 * WordPress dependencies
 */
import { Path, SVG } from '@wordpress/primitives';

export const FlickrIcon = () => (
	<SVG width="24" height="24" viewBox="0 0 24 24" version="1.1">
		<Path d="M6.5,7c-2.75,0-5,2.25-5,5s2.25,5,5,5s5-2.25,5-5S9.25,7,6.5,7z M17.5,7c-2.75,0-5,2.25-5,5s2.25,5,5,5s5-2.25,5-5 S20.25,7,17.5,7z" />
	</SVG>
);
