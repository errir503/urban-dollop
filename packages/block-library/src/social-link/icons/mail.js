/**
 * WordPress dependencies
 */
import { Path, SVG } from '@wordpress/primitives';

export const MailIcon = () => (
	<SVG width="24" height="24" viewBox="0 0 24 24" version="1.1">
		<Path d="M19 5H5c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V7c0-1.1-.9-2-2-2zm.5 12c0 .3-.2.5-.5.5H5c-.3 0-.5-.2-.5-.5V9.8l7.5 5.6 7.5-5.6V17zm0-9.1L12 13.6 4.5 7.9V7c0-.3.2-.5.5-.5h14c.3 0 .5.2.5.5v.9z" />
	</SVG>
);
