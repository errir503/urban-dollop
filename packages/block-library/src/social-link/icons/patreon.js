/**
 * WordPress dependencies
 */
import { Circle, Rect, SVG } from '@wordpress/primitives';

export const PatreonIcon = () => (
	<SVG width="24" height="24" viewBox="0 0 569 546" version="1.1">
		<Circle cx="363" cy="205" r="205" />
		<Rect width="100" height="546" x="0" y="0" />
	</SVG>
);
