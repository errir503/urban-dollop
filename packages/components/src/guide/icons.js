/**
 * WordPress dependencies
 */
import { SVG, Circle } from '@wordpress/primitives';

export const PageControlIcon = ( { isSelected } ) => (
	<SVG width="8" height="8" fill="none" xmlns="http://www.w3.org/2000/svg">
		<Circle
			cx="4"
			cy="4"
			r="4"
			fill={ isSelected ? '#419ECD' : '#E1E3E6' }
		/>
	</SVG>
);
