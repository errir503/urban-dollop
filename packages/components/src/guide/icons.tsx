/**
 * WordPress dependencies
 */
import { SVG, Circle } from '@wordpress/primitives';

export const PageControlIcon = () => (
	<SVG width="8" height="8" fill="none" xmlns="http://www.w3.org/2000/svg">
		<Circle cx="4" cy="4" r="4" />
	</SVG>
);
