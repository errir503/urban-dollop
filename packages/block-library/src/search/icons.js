/**
 * WordPress dependencies
 */
import { SVG, Rect } from '@wordpress/components';

export const buttonOnly = (
	<SVG xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">
		<Rect x="7" y="10" width="10" height="4" rx="1" fill="currentColor" />
	</SVG>
);

export const buttonOutside = (
	<SVG xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">
		<Rect
			x="4.75"
			y="15.25"
			width="6.5"
			height="9.5"
			transform="rotate(-90 4.75 15.25)"
			stroke="currentColor"
			strokeWidth="1.5"
			fill="none"
		/>
		<Rect x="16" y="10" width="4" height="4" rx="1" fill="currentColor" />
	</SVG>
);

export const buttonInside = (
	<SVG xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">
		<Rect
			x="4.75"
			y="15.25"
			width="6.5"
			height="14.5"
			transform="rotate(-90 4.75 15.25)"
			stroke="currentColor"
			strokeWidth="1.5"
			fill="none"
		/>
		<Rect x="14" y="10" width="4" height="4" rx="1" fill="currentColor" />
	</SVG>
);

export const noButton = (
	<SVG xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">
		<Rect
			x="4.75"
			y="15.25"
			width="6.5"
			height="14.5"
			transform="rotate(-90 4.75 15.25)"
			stroke="currentColor"
			fill="none"
			strokeWidth="1.5"
		/>
	</SVG>
);

export const buttonWithIcon = (
	<SVG xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">
		<Rect
			x="4.75"
			y="7.75"
			width="14.5"
			height="8.5"
			rx="1.25"
			stroke="currentColor"
			fill="none"
			strokeWidth="1.5"
		/>
		<Rect x="8" y="11" width="8" height="2" fill="currentColor" />
	</SVG>
);

export const toggleLabel = (
	<SVG xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">
		<Rect
			x="4.75"
			y="17.25"
			width="5.5"
			height="14.5"
			transform="rotate(-90 4.75 17.25)"
			stroke="currentColor"
			fill="none"
			strokeWidth="1.5"
		/>
		<Rect x="4" y="7" width="10" height="2" fill="currentColor" />
	</SVG>
);
