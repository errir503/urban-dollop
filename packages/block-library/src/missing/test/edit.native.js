/**
 * External dependencies
 */
import renderer from 'react-test-renderer';
import { Text } from 'react-native';

/**
 * WordPress dependencies
 */
import { BottomSheet, Icon } from '@wordpress/components';
import { help, plugins } from '@wordpress/icons';
import { storeConfig } from '@wordpress/block-editor';
jest.mock( '@wordpress/blocks' );
jest.mock( '@wordpress/block-editor/src/store/selectors' );

/**
 * Internal dependencies
 */
import UnsupportedBlockEdit from '../edit.native.js';

const defaultAttributes = {
	originalName: 'missing/block/title',
};

const getTestComponentWithContent = ( attributes = defaultAttributes ) => {
	return renderer.create(
		<UnsupportedBlockEdit attributes={ attributes } />
	);
};

describe( 'Missing block', () => {
	beforeEach( () => {
		storeConfig.selectors.getSettings.mockReturnValue( {} );
	} );

	it( 'renders without crashing', () => {
		const component = getTestComponentWithContent();
		const rendered = component.toJSON();
		expect( rendered ).toMatchSnapshot();
	} );

	describe( 'help modal', () => {
		it( 'renders help icon', () => {
			const component = getTestComponentWithContent();
			const testInstance = component.root;
			const icons = testInstance.findAllByType( Icon );
			expect( icons.length ).toBe( 2 );
			expect( icons[ 0 ].props.icon ).toBe( help );
		} );

		it( 'renders info icon on modal', () => {
			const component = getTestComponentWithContent();
			const testInstance = component.root;
			const bottomSheet = testInstance.findByType( BottomSheet );
			const children = bottomSheet.props.children[ 0 ].props.children;
			expect( children.length ).toBe( 3 ); // 4 children in the bottom sheet: the icon, the "isn't yet supported" title and the "We are working hard..." message
			expect( children[ 0 ].props.icon ).toBe( help );
		} );

		it( 'renders unsupported text on modal', () => {
			const component = getTestComponentWithContent();
			const testInstance = component.root;
			const bottomSheet = testInstance.findByType( BottomSheet );
			const children = bottomSheet.props.children[ 0 ].props.children;
			expect( children[ 1 ].props.children ).toBe(
				"'" +
					defaultAttributes.originalName +
					"' is not fully-supported"
			);
		} );

		describe( 'Unsupported block editor (UBE)', () => {
			beforeEach( () => {
				// By default we set the web editor as available
				storeConfig.selectors.getSettings.mockReturnValue( {
					unsupportedBlockEditor: true,
				} );
			} );

			it( 'renders edit action if UBE is available', () => {
				const component = getTestComponentWithContent();
				const testInstance = component.root;
				const bottomSheet = testInstance.findByType( BottomSheet );
				const bottomSheetCells = bottomSheet.props.children[ 1 ];
				expect( bottomSheetCells ).toBeTruthy();
				expect( bottomSheetCells.props.children.length ).toBe( 2 );
				expect( bottomSheetCells.props.children[ 0 ].props.label ).toBe(
					'Edit using web editor'
				);
			} );

			it( 'does not render edit action if UBE is not available', () => {
				storeConfig.selectors.getSettings.mockReturnValue( {
					unsupportedBlockEditor: false,
				} );

				const component = getTestComponentWithContent();
				const testInstance = component.root;
				const bottomSheet = testInstance.findByType( BottomSheet );
				expect( bottomSheet.props.children[ 1 ] ).toBeFalsy();
			} );

			it( 'does not render edit action if the block is incompatible with UBE', () => {
				const component = getTestComponentWithContent( {
					originalName: 'core/block',
				} );
				const testInstance = component.root;
				const bottomSheet = testInstance.findByType( BottomSheet );
				expect( bottomSheet.props.children[ 1 ] ).toBeFalsy();
			} );
		} );
	} );

	it( 'renders admin plugins icon', () => {
		const component = getTestComponentWithContent();
		const testInstance = component.root;
		const icons = testInstance.findAllByType( Icon );
		expect( icons.length ).toBe( 2 );
		expect( icons[ 1 ].props.icon ).toBe( plugins );
	} );

	it( 'renders title text without crashing', () => {
		const component = getTestComponentWithContent();
		const testInstance = component.root;
		const texts = testInstance.findAllByType( Text );
		expect( texts.length ).toBe( 2 );
		expect( texts[ 0 ].props.children ).toBe( 'missing/block/title' );
		expect( texts[ 1 ].props.children ).toBe( 'Unsupported' );
	} );
} );
