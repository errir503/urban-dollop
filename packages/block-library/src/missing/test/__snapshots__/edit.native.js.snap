// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Missing block renders without crashing 1`] = `
<View
  accessibilityHint="Tap here to show help"
  accessibilityLabel="Help button"
  accessibilityRole="button"
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
>
  <View
    accessibilityHint="Tap here to show help"
    accessibilityLabel="Help button"
    accessibilityRole="button"
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={null}
  >
    Svg
  </View>
  Svg
  <Text>
    missing/block/title
  </Text>
  <Text>
    Unsupported
  </Text>
</View>
`;
